grafana/grafana
# Elasticsearch: Fix processing of response with multiple group by for alerting

This PR fixes processing of response with multiple group by on backend. On backend, we were missing using fields from frames if there already were any, and add fields from `propKeys` if there weren't any. This is correctly done on [frontend implementation](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/ElasticResponse.ts#L179) where we check if we have any `columns` and if yes, we use them instead of adding `propKeys`. 

Fixes: https://github.com/grafana/grafana/issues/64727
Part of #54011

How to test:
1. I added test for this scenario, so first of all, it should pass 🙂
2. With  feature toggle `elasticsearchBackendMigration=true` run `make devenv sources=elastic`
3. Open Explore and run query following query with [2 terms group by](http://localhost:3000/explore?orgId=1&left=%7B%22datasource%22:%22PE50363A9B6833EE7%22,%22queries%22:%5B%7B%22alias%22:%22%22,%22bucketAggs%22:%5B%7B%22field%22:%22label%22,%22id%22:%222%22,%22settings%22:%7B%22min_doc_count%22:%221%22,%22order%22:%22desc%22,%22orderBy%22:%22_term%22,%22size%22:%2210%22%7D,%22type%22:%22terms%22%7D,%7B%22field%22:%22level%22,%22id%22:%223%22,%22settings%22:%7B%22min_doc_count%22:%221%22,%22order%22:%22desc%22,%22orderBy%22:%22_term%22,%22size%22:%2210%22%7D,%22type%22:%22terms%22%7D%5D,%22datasource%22:%7B%22type%22:%22elasticsearch%22,%22uid%22:%22PE50363A9B6833EE7%22%7D,%22metrics%22:%5B%7B%22field%22:%22counter%22,%22id%22:%221%22,%22type%22:%22max%22%7D%5D,%22query%22:%22%22,%22refId%22:%22A%22,%22timeField%22:%22@timestamp%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D)

<img width="1431" alt="image" src="https://user-images.githubusercontent.com/30407135/226873270-fe7cc1e3-e9f5-497e-aac4-9b482001385d.png">
4. There should be a correct response in table
